### ISSUE TITLE
VoiceOver sometimes skips release alert of balloon

### ISSUE THREAD

**Issue by jessegreenberg** [phetsims/balloons-and-static-electricity#373]
Often the release alert is missing in VoiceOver. It is possibly struggling with change of focus (draggable balloon to balloon button, or draggable balloon to wall button). Or it could be a performance problem because much of the DOM changes when the balloon is released.

I am considering a workaround where we delay all alerts by the utteranceQueue interval as a potential workaround for this. At the moment, the utteranceQueue is on a continuous interval so delay between change of focus and next utterance could be just a couple of ms.

**Comment by jessegreenberg** [comment]
I tried changing the implementation of utteranceQueue so that it it delays by interval for every alert. I increased the interval to  allow focus to  move well before an alert is triggered. It didn't help much. In the case where the balloon is, many alerts stop until the balloon hits its destination sweater. Presumably this is because there are many DOM updates happening at once and Safari/VO is bogging down with rapid changes.

We should test to see if disabling updates to dynamic descsriptions fixes the problem. If so, it is time to work on batching changes to accessible labels and descriptions.

**Comment by jessegreenberg** [comment]
I have been playing with this. A few notes:

- I tried short circuiting setTextContent in Accessibility.js to no-op, and it had no impact on this issue. So this is not a performance problem caused by DOM updates.
- When I closed many browser tabs I had open (~10) performance of VO substantially increased. This is no longer a problem for the mac I am testing on.
- About 60% of the time,  we miss the first "relesed" announcement. Since we hear the remaining "Moves quickly left..." alerts, I do not think this is related to performance, but because VO has a hard time reading the first alert after focus changes.

**Comment by jessegreenberg** [comment]
> About 60% of the time, we miss the first "relesed" announcement. Since we hear the remaining "Moves quickly left..." alerts, I do not think this is related to performance, but because VO has a hard time reading the first alert after focus changes.

This is an AT behavior, and I am not sure if it is worth spending more time on this. Since performance of VoiceOver has improved with far fewer tabs open, I am going to close this issue.
